### ISSUE TITLE
[Bug] False warning of not loaded .py modules after `save` operation

### ISSUE THREAD

**Issue by CSharperMantle** [rapid7/metasploit-framework#12943]
#### Step 1: Select a python module

Start a msfconsole and select **any** python module. In this issue, module `exploit/windows/smb/ms17_010_eternalblue_win8` is used. The module file itself on disk is `/usr/share/metasploit-framework/modules/exploits/windows/smb/ms17_010_eternalblue_win8.py`.

After this step, the console window will have the following output.
```plain
mantlebao@mantlekali:~$ msfconsole
[ BANNER OMITTED ]
msf5 exploit(linux/redis/redis_unauth_exec) > use exploit/windows/smb/ms17_010_
use exploit/windows/smb/ms17_010_eternalblue
use exploit/windows/smb/ms17_010_eternalblue_win8
use exploit/windows/smb/ms17_010_psexec
msf5 exploit(linux/redis/redis_unauth_exec) > use exploit/windows/smb/ms17_010_eternalblue_win8 
msf5 exploit(windows/smb/ms17_010_eternalblue_win8) >
```
#### Step 2: Run `save` command

Run `save` command, and then exit.
Console output:
```plain
msf5 exploit(windows/smb/ms17_010_eternalblue_win8) > save
Saved configuration to: /home/mantlebao/.msf4/config
msf5 exploit(windows/smb/ms17_010_eternalblue_win8) > exit
mantlebao@mantlekali:~$
```

#### Step 3: Re-enter the console
Type `msfconsole` to start a new console. Notice the false warning given and the module which is actually loaded successfully.
```plain
mantlebao@mantlekali:~$ msfconsole
[-] WARNING! The following modules could not be loaded!
[-]     /usr/share/metasploit-framework/modules/exploits/windows/smb/ms17_010_eternalblue_win8.rb
[-] Please see /home/mantlebao/.msf4/logs/framework.log for details.
                                                  
[ BANNER OMITTED ]

msf5 exploit(windows/smb/ms17_010_eternalblue_win8) >
```

## Expected behavior

No warnings should come out.

## Current behavior

False warning is given.

## Critical log
```plain
[02/17/2020 21:12:13] [e(0)] core: /usr/share/metasploit-framework/modules/exploits/windows/smb/ms17_010_eternalblue_win8.rb failed to load due to the following error: Errno::ENOENT No such file or directory @ rb_sysopen - /usr/share/metasploit-framework/modules/exploits/windows/smb/ms17_010_eternalblue_win8.rb Call stack: /usr/share/metasploit-framework/lib/msf/core/modules/loader/directory.rb:77:in `initialize' /usr/share/metasploit-framework/lib/msf/core/modules/loader/directory.rb:77:in `open' /usr/share/metasploit-framework/lib/msf/core/modules/loader/directory.rb:77:in `read_module_content' /usr/share/metasploit-framework/lib/msf/core/modules/loader/base.rb:127:in `load_module' /usr/share/metasploit-framework/lib/msf/core/module_manager/cache.rb:91:in `block in load_cached_module' /usr/share/metasploit-framework/lib/msf/core/module_manager/cache.rb:86:in `each' /usr/share/metasploit-framework/lib/msf/core/module_manager/cache.rb:86:in `load_cached_module' /usr/share/metasploit-framework/lib/msf/core/module_set.rb:45:in `create' /usr/share/metasploit-framework/lib/msf/core/module_manager.rb:90:in `create' /usr/share/metasploit-framework/lib/msf/ui/console/command_dispatcher/modules.rb:667:in `cmd_use' /usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:523:in `run_command' /usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:474:in `block in run_single' /usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:468:in `each' /usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:468:in `run_single' /usr/share/metasploit-framework/lib/msf/ui/console/driver.rb:278:in `block in load_config' /usr/share/metasploit-framework/lib/msf/ui/console/driver.rb:275:in `each_pair' /usr/share/metasploit-framework/lib/msf/ui/console/driver.rb:275:in `load_config' /usr/share/metasploit-framework/lib/msf/ui/console/driver.rb:171:in `initialize' /usr/share/metasploit-framework/lib/metasploit/framework/command/console.rb:62:in `new' /usr/share/metasploit-framework/lib/metasploit/framework/command/console.rb:62:in `driver' /usr/share/metasploit-framework/lib/metasploit/framework/command/console.rb:48:in `start' /usr/share/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start' /usr/bin/msfconsole:49:in `<main>'
```

### Metasploit version
```plain
Framework: 5.0.72-dev
Console  : 5.0.72-dev
```

### I installed Metasploit with:
- [x] Kali package via apt
- [ ] Omnibus installer (nightly)
- [ ] Commercial/Community installer (from http://www.rapid7.com/products/metasploit/download.jsp)
- [ ] Source install (please specify ruby version)

### OS
```plain
Distributor ID: Kali
Description:    Kali GNU/Linux Rolling
Release:        2020.1
Codename:       kali-rolling
```

**Comment by CSharperMantle** [comment]
Reply.

**Comment by CSharperMantle** [comment]
I have (hopefully) figured out why this issue happened and will open a pull request very soon.

https://github.com/rapid7/metasploit-framework/blob/5dd6d1b4b287384cb00c1c0091f497a9d630e9f2/lib/msf/core/module_manager/cache.rb#L84-L93

These lines of code will iterate over a list of loaders ( [`Msf::Modules::Loader::Directory`](https://github.com/rapid7/metasploit-framework/blob/5dd6d1b4b287384cb00c1c0091f497a9d630e9f2/lib/msf/core/modules/loader/directory.rb) and [`Msf::Modules::Loader::Executable`](https://github.com/rapid7/metasploit-framework/blob/5dd6d1b4b287384cb00c1c0091f497a9d630e9f2/lib/msf/core/modules/loader/executable.rb) currently) to try to load modules.

The `Directory` loader mainly focuses on loading Ruby modules so it will fail to load `exploit/windows/smb/ms17_010_eternalblue_win8` as it is a Python module. Normally the loading job will then be passed to `Executable` loader, which *can* correctly load Python modules, by the main loop. However, *these two loaders manage their own error reporting mechanisms*, which means the `Directory` loader will report a failed attempt of loading, not knowing that the `Executable` loader can actually load it.

https://github.com/rapid7/metasploit-framework/blob/5dd6d1b4b287384cb00c1c0091f497a9d630e9f2/lib/msf/core/modules/loader/directory.rb#L81-L83

The upcoming PR will fix the issue by upbringing the error reporting to the main loop, after the `ModuleManager` finds out that *no* loader can correctly load the module.

**Comment by CSharperMantle** [comment]
Fixed by #14480.
